corehq/motech/repeaters/models.py

```diff
@@ -1185,7 +1185,8 @@ def failure_reason(self):
     def last_message(self):
         # Uses `list(queryset)[-1]` instead of `queryset.last()` to use
         # prefetched attempts, if available.
-        return list(self.attempts)[-1].message
+        attempts = list(self.attempts)
+        return attempts[-1].message if attempts else None
```

Comment:
Nit: could this return an empty string rather than `None` in the case of no attempts? That would make the return type always be the same (requiring less `if x is None` checks elsewhere) without breaking `process(message) if message else ignore`.

Reply:
I thought about this, and those are good points.

I leaned towards `None` because `attempts[-1].message` could be `None` anyway:
```python
    message = models.TextField(null=True, blank=True)
```

I could return a string always:
```python
    def last_message(self):
        attempts = list(self.attempts)
        if attempts
            return attempts[-1].message or ''
        return ''
```

Or I could warn future devs to take `None` into consideration -> :wink: :
```python
    def last_message(self) -> Optional[str]:
        attempts = list(self.attempts)
        return attempts[-1].message if attempts else None
```

I don't really love the idea of changing the model. Semantically `None` means that message has not been assigned a value. I that's useful to know, and maybe `last_message()` should convey that. What do you think?

Reply:
I think up to you. I'd personally go with a `NOT NULL` field on the model and make it always return a string since that tends to make many things easier and less error prone in my experience. Is there ever a time when it's important to know that a message has not been assigned (value is `None`) vs there is no message (value is empty string)?

Reply:
> Is there ever a time when it's important to know that a message has not been assigned

An excellent question. e686887ca7fa9674b3af240c29cfa4b860dd6dc6